src/Schema/Comparator.php

```diff
@@ -228,12 +248,28 @@ public function diffTable(Table $fromTable, Table $toTable)
             }
 
             // See if column has changed properties in "to" table.
-            $changedProperties = $this->diffColumn($column, $toTable->getColumn($columnName));
-
-            if (count($changedProperties) === 0) {
+            if (! $this->compareColumns($column, $toTable->getColumn($columnName))) {
                 continue;
             }
 
+            // Backwards compatibility
+            if ($this->platform === null) {
+                $changedProperties = $this->diffColumn($column, $toTable->getColumn($columnName));
+            } else {
+                $properties1 = $column->toArray();
+                $properties2 = $toTable->getColumn($columnName)->toArray();
+
+                $changedProperties = [];
+
+                foreach (array_keys(array_merge($properties1, $properties2)) as $key) {
+                    if (($properties1[$key] ?? null) === ($properties2[$key] ?? null)) {
```

Comment:
Why do we still need to do a property-based comparison while the platform is available?